Frends.CSV.Create/Frends.CSV.Create/Create.cs

```
@@ -90,43 +90,73 @@ private static string ListToCsvString(List<List<object>> inputData, List<string>
         return csvString.ToString();
     }
 
-    private static string JsonToCsvString(string json, CsvConfiguration config, Options options, CancellationToken cancellationToken)
+    private static string JsonToCsvString(string json, Input input, CsvConfiguration config, Options options, CancellationToken cancellationToken)
     {
         json = StringifyJsonNumbers(json);
-        JToken input = JsonConvert.DeserializeObject<JToken>(
+        JToken jToken = JsonConvert.DeserializeObject<JToken>(
             json,
             new JsonSerializerSettings { DateParseHandling = DateParseHandling.None, }
         );
-        input = input is JObject ? new JArray() { input } : input;
-        JArray inputArray = (JArray) input;
-        var limits = JsonArraysLimits(inputArray);
-        var data = FlattenJson(inputArray, limits);
+        JArray inputArray = jToken is JObject ? new JArray { jToken } : (JArray)jToken;
 
         //CSV part
         using var csvString = new StringWriter();
         using var csv = new CsvWriter(csvString, config);
 
-        //Write the header row
-        if (config.HasHeaderRecord && data.Any())
+        //If headers are specified manually
+        if (input.SpecifyHeadersManually && input.ManualHeaders != null && input.ManualHeaders.Any())
```

so, if manual headers are empty or null - we should probably just throw an error message about it.